bootstrap/kubeadm/main.go

```diff
@@ -202,7 +202,7 @@ func main() {
 		os.Exit(1)
 	}
 
-	diagnosticsOpts := flags.GetDiagnosticsOptions(diagnosticsOptions)
```

Comment:
Hm all of this becomes a bit messy. On one side we should update the description of the tls flags in the GetTLSOptionOverrideFuncs util. On the other side if we add there that these options also affect the metrics server this is only true if folks actually don't forget to pass in the tlsOptionOverrides into the GetDiagnosticsOptions func

Reply:
Makes me wonder if it's still a good idea to have two separate flag util funcs if they actuallyl only make sense if they are used together

Reply:
What if we merge the two funcs and option structs and just call them flags.ManagerOptions?

Reply:
@chrischdi @awprice WDYT?

Reply:
(would be nice to get this in quickly for the next CAPI release)

Reply:
+1 for merging them